src/controllers/transactions/repurchaseController.js

```diff
@@ -6,25 +6,22 @@ export const convertAndCreateRepurchaseStockOnchain = async (
     { stakeholderId,
         stockClassId,
         security_id,
-        considertationText= "",
+        considertationText = "",
         quantity,
         price,
         comments = [] }
 ) => {
-    const secIdBytes16 = convertUUIDToBytes16(security_id);
-    const stakeHolderIdBytes16 = convertUUIDToBytes16(stakeholderId);
-    const stockClassIdBytes16 = convertUUIDToBytes16(stockClassId);
-    const scaledQuantity = toScaledBigNumber(quantity)
-    const scaledPrice =  toScaledBigNumber(price.amount)
+    const scaledPrice = toScaledBigNumber(price.amount)
 
-    const tx = await contract.repurchaseStock(
-        stakeHolderIdBytes16,
-        stockClassIdBytes16,
-        secIdBytes16,
+    const tx = await contract.repurchaseStock({
+        stakeholder_id: convertUUIDToBytes16(stakeholderId),
+        stock_class_id: convertUUIDToBytes16(stockClassId),
+        security_id: convertUUIDToBytes16(security_id),
+        quantity: toScaledBigNumber(quantity),
         comments,
-        considertationText,
-        scaledQuantity,
-        scaledPrice
-    );
+        nonce: 0, // needed because of StockParamsQuantity Struct
```

Comment:
@apfong, could you clarify the need for nonce in the StockParamsQuantity struct, used in repurchaseStock and cancelStock? Is it possible to remove it? Alternatively, I considered passing nonce as a function argument in repurchaseStockByTA, but I'm unsure about the original design intentions. Your insights would be helpful.
```solidity   
 function repurchaseStockByTA(
        StockParamsQuantityWithoutNonce memory params,
        uint256 price,
        ActivePositions storage positions,
        SecIdsStockClass storage activeSecs,
        address[] storage transactions,
        Issuer storage issuer,
        StockClass storage stockClass,
        uint256 nonce
    ) external {
```

Reply:
Yeah can try to clarify, lmk if this makes sense. Basically I was just trying to keep the functionality the same whenever possible and introduced some structs where there seemed to be common groupings of params. For most the regular StockParams contained all the required params, but in a few cases there were additional quantity and nonce variables and so I created the StockParamsQuantity to use in those cases.

If you've found that you can just use StockParams and pass in the nonce separately without running into the via-ir stack limits then I don't see any problem with removing the nonce from the quantity struct

Reply:
Thanks for the clarification.